pages/_app.tsx

```diff
@@ -1,13 +1,48 @@
-import "../styles/globals.css";
+/* --- libs --------------------------------------------------------------------------------------------------------- */
 import type { AppProps } from "next/app";
+import { RecoilRoot, useSetRecoilState } from "recoil";
+import { useEffect } from "react";
+import { parseCookies } from "nookies";
+
+/* --- globalState --------------------------------------------------------------------------------------------------- */
+import { Cart, cartState } from "../stores/cart";
+
+/* --- assets ------------------------------------------------------------------------------------------------------- */
+import "../styles/globals.css";
+
+/* --- components --------------------------------------------------------------------------------------------------- */
 import { Header } from "../components/layouts/Header/Header";
-import { RecoilRoot } from "recoil";
+
+/**
+ * リロードした際にglobalStateがリセットされるので、Cookiesに値が存在すればglobalStateを更新する
+ * RecoilRoot内ではないとuseSetRecoilStateが使えない為、別関数で定義
+ */
+function InitApp() {
+  const setCart = useSetRecoilState(cartState);
+
+  useEffect(() => {
+    const cookies = parseCookies();
+    let cookiesCart: Cart;
+
+    if ({ cookies }.cookies.cart !== undefined) {
```

Comment:
こちら少し冗長な書き方になっていそうです。
```suggestion
    if (cookies.cart !== undefined) {
```

Reply:
ありがとうございます！
こちら修正いたしました！